src/vision/frc_vision_final.py

```diff
@@ -0,0 +1,190 @@
+
+
+def newLine(filtered_lines, new_line, filtered_line_img, x1, y1, x2, y2):
+    filtered_lines.append(new_line)
+    cv2.line(filtered_line_img, (x1, y1), (x2, y2), (0, 255, 0), 1)
+    return x1 + x2, y1, y2
+
+
+NetworkTables.initialize(server='roborio-166-frc.local')
+
+sd = NetworkTables.getTable('SmartDashboard')
+# sd.putNumber('someNumber', 1234)
+# otherNumber = sd.getNumber('otherNumber')
+
+# Resolution
+WIDTH = 640
+HEIGHT = 480
+POINT_SAMPLES = 5
+# Angle value of each pixel
+FOV_ANGLE = 82.5
+PIXEL_ANGLE = FOV_ANGLE/WIDTH
+
+# Camera settings
+pipe = rs2.pipeline()
+config = rs2.config()
+config.enable_stream(rs2.stream.color, WIDTH, HEIGHT, rs2.format.bgr8, 30)
+config.enable_stream(rs2.stream.depth, WIDTH, HEIGHT, rs2.format.z16, 30)
+profile = pipe.start(config)
+s = profile.get_device().query_sensors()[1]
+s.set_option(rs2.option.brightness, 0)
+s.set_option(rs2.option.contrast, 100)
+s.set_option(rs2.option.exposure, 45)
+s.set_option(rs2.option.gain, 75)
+s.set_option(rs2.option.gamma, 100)
+s.set_option(rs2.option.hue, 0)
+s.set_option(rs2.option.saturation, 50)
+s.set_option(rs2.option.sharpness, 0)
+s.set_option(rs2.option.white_balance, 2800)
+
+X_VALS = []
+Y_VALS = []
+
+POINTER = 0
+
+while True:
+    start_time = time.time()
+
+    # Waits to get frames, and gets information about the frame.
+    frames = rs2.composite_frame(pipe.wait_for_frames())
+    frame = rs2.video_frame(frames.get_color_frame())
+    depth = rs2.depth_frame(frames.get_depth_frame())
+
+    if not frame:
+        continue
+
+    IMG = np.asanyarray(frame.get_data())
+
+    # Convert from RGB to HSV, helps with filtering
+    HSV = cv2.cvtColor(IMG, cv2.COLOR_BGR2HSV)
+
+    # Define upper and lower bounds for HSV variables
+    LOWER_COLOR = np.array([70, 80, 255])
+    UPPER_COLOR = np.array([95, 180, 255])
+
+    # Create mask within hsv range
+    MASK = cv2.inRange(HSV, LOWER_COLOR, UPPER_COLOR)
+
+    # Various blur method testings (buffer for pixel imperfections)
+    BLUR = cv2.GaussianBlur(MASK, (3, 3), 0)
+    MEDIAN = cv2.medianBlur(MASK, 3)
+
+    # Edge detection on each test for use in line detection
+    BLUR_EDGES = cv2.Canny(BLUR, 100, 200)
+    MASK_EDGES = cv2.Canny(MASK, 100, 200)
+    MED_EDGES = cv2.Canny(MEDIAN, 50, 150)
+
+    # Empty image for drawing lines (testing)
+    FILTERED_LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
+    LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
+
+    # Find lines in selected image
+    LINES = cv2.HoughLinesP(MASK_EDGES, 1, radians(.5), 25, maxLineGap=25)
+
+    # If there are lines:
+    if LINES is not None:
+        NUM_LINES = len(LINES)
+        FILTERED_LINES = []
+        X_TOTAL = 0
+        Y_TOTAL = 0
+
+        for NEW_LINE in LINES:
+            x1, y1, x2, y2 = NEW_LINE[0]
+            new_slope = degrees(np.arctan((y2 - y1)/(x2 - x1)))
+
+            # Checks if we have verified lines, and makes a new line based on that.
+            if FILTERED_LINES:
```

Comment:
At this point, `FILTERED_LINES` is always going to be false.

Reply:
In testing, it does run the code in here, and without it the code crashes

Reply:
Wait, really? Where is `FILTERED_LINES` being given a value then?

Reply:
Found it, line 126